### ISSUE TITLE
Low gain in Phone Mic

### ISSUE THREAD

**Issue by ManasGhosh123** [PixelExperience/android-issues#3167]
## Build date

PixelExperience_Plus_violet-12.1-20220718-0524-OFFICIAL

## Expected Behavior

The primary microphone on the phone should pick up better voice gain during calls

## Current Behavior

The person on the other side of the call can barely hear me. Same problem occurs during voice recording. The situation is better while using secondary mic or headphones.

## Possible Solution

Increase the gain from the primary microphone of the device.

## Steps to Reproduce

1. Make a call using the device without any secondary mic or headphones.
2. Ask the other side person for the voice quality and loudness compared to other calls.
3. Record a voice clip using voice recorder.
4. hear it back to assess.

/device violet
/version twelve_plus

**Comment by pixelexperiencebot** [comment]
Issue created! You can close at any time by commenting ```/close```

**Comment by freethud** [comment]
The same problem with two my devices:

1. Xiaomi Redmi Note 5 pro (whyred) PixelExperience_Plus_whyred-12.1-20220719-0603-OFFICIAL.zip
2. Xiaomi Redmi Note 7 (lavender) PixelExperience_Plus_lavender-12.1-20220720-0644-OFFICIAL.zip

When I'm calling anyone, the volume is very very low!! I'm non able to hear...
Have you got any solution???
Thank you !

**Comment by goofy84** [comment]
Same issue with a Xiaomi mi8 Low Volume during calls. I was on Pixel experince 12 plus and now on 13 Beta.

**Comment by Dionysus74** [comment]
I had the same problem with my OnePlus 7 Pro. Low call quality and/or choppy call and recording quality so I rolled back to stock OxygenOS.

**Comment by mohammedattm** [comment]
Same problem
REDMI NOTE 9 PRO microphone doesn't work on loud speaker

**Comment by WakoKnight** [comment]
I have a Pixel 3 and have a problem with the main microphone (the front one, not the one on top).

I tested both microphones with Microphone Amplifier that show sound waves in real time. Just open the app, press on “Enter Amplifier”, put gain slider around 50% and then press the power button on the bottom centre. When using Mic 1 in the app (the button before the “Rec” button), both microphones are used. You just need to move your finder on top of each microphone to see they are active and see the sound waves in real time. When switching to Mic 2 in the app, only the front microphone is used at about the same level of sound. That confirms they both work fine.

Then I tested with the Google Recorder app where only the front microphone is used (still moving my finger on both microphones to see if they are used or not). The sound is recorded clear and loud.

The problem happens when I make a call using the front microphone (not in speakerphone mode). Peoples say they almost don’t hear me unless I speak really loud, almost yelling. There seem to be a problem only with the phone app, all other applications work well with the main microphone. I did add all authorizations to the phone app.

I did call my cell phone number from my Google Home and I need to talk really loud to hear myself back when using the front microphone. When on speakerphone mode, the sound is loud and clear. I then called a VOIP app on my phone and then the front microphone was working fine (loud and clear). So this confirms the problem happens in the phone app only (for the Pixel 3 I have).

I don’t know if it has something to do, but my phone is detected as a Nexus 6P in my Google account, but as a Pixel 3 on Google Play and in phone settings.

I hope a solution will be found soon for the phone app.

**Comment by WakoKnight** [comment]
I installed some ROMs and here is the conclusion:

Android 12 (factory image from Google): Microphone is working fine when I call someone.
Pixel Experience 12: No tested.
Pixel Experience Plus 12: Microphone is working fine when I call someone.
Pixel Experience 13: Microphone is NOT working fine when I call someone. People I call have trouble to hear me.
Pixel Experience Plus 12: Microphone is NOT working fine when I call someone. People I call have trouble to hear me.

It seems, there is something to fix in the Pixel Experience 13 ROMs for it to work on Pixel 3 phones.

**Comment by HAMZterr** [comment]
Redmi note 9 pro.
The other person get very poor and low voice quality from my side.. while using microphone it works fine.
Had same issue in previous update and now facing again in march.

**Comment by ARYxN04** [comment]
I got it fixed recently after scrolling through endless forums and comment sections:
Ok, so I first removed both sims with the phone off. Then i put one of my 2 sims ONLY in slot 2 like you said. I inserted it into the phone, booted it up, and made a test call. To my surprise, the phone wasn't on speaker. I couldn't beileve it. I tried again, and it worked. I switched to speaker, now it's on speaker. I turned off the speaker and it was back to the earpeice! I tried whatsapp, and same result, the earpiece was working. I then decided to shut the phone down, and put the both sim 1 and 2 back in the phone. Guess what? after booting the phone up again, and making a test call, it still works! I don't understand the rhyme or reason behind it, but I'm not complaining. I'm so happy it FINALLY works.

to recap, Basically here are the steps:
1. Turn Off phone
2. Remove both sim cards.
3. Put a sim only in slot 2
4. Turn on phone and make a test call, it should no longer be on speaker!
5. go ahead and shut down the phone, and put a sim in both slots, if you are using 2 sims.
6. Turn the phone back on, and make a test call. It should still be working on the earpiece!


credit: https://xdaforums.com/t/redmi-note-9-pro-speakerphone-is-always-active-on-custom-roms-cant-switch-to-earpiece.4390157/post-86442419
